samtools.wdl

```diff
@@ -32,10 +33,12 @@ task Index {
         File bamFile
         String bamIndexPath
 
-        String dockerTag = "1.8--h46bd0b3_5"
+        String dockerImage = "quay.io/biocontainers/samtools:1.8--h46bd0b3_5"
     }
 
     command {
+        set -e
+        mkdir -p $(dirname ~{bamIndexPath})
```

Comment:
In practice this directory should always already exist, as the index should always be in the same directory as the BAM file.

Reply:
Yeah but there is also file localization with cromwell. It gets a little hairy. This index task should not be used anyway since it does not couple the bamfile and its index. I thought I had tackled this by doing some stuff with linking .:thinking:

Reply:
Linking is hairy when run from a container, at least soft-linking is. I think localization issues should be solved here by returning the input BAM file as an output as well and ensuring that the index is placed in the same directory as the input BAM.

Reply:
I solved it with a hardlink now. Should cover most use cases... If you agree of course. Otherwise I will try indexing the file in `inputs` but that might require some complex bash stuff.